writers_rollingfilewriter.go

```diff
@@ -373,8 +373,8 @@ func (rw *rollingFileWriter) archiveExplodedLogs(logFilename string, compression
 		}
 
 		// Finalize archive by swapping the buffered archive into place
-		archiveFile := path.Clean(rw.archivePath + "/" + compressionType.rollingArchiveTypeName(logFilename, true))
```

Comment:
I decided to clean this up, because this file was inconsistent between `path` and `filepath`. Since I'm modifying it, I might as well update it to be idiomatic.